File: app/src/main/java/com/rendersoncs/reportform/view/services/util/AlertDialogUtil.java
package com.rendersoncs.reportform.view.services.util;

import android.app.Activity;
import android.content.DialogInterface;

import androidx.appcompat.app.AlertDialog;

public class AlertDialogUtil {


    public void showDialog(Activity activity, String title, String msg, String positiveLabel,
                           DialogInterface.OnClickListener positiveOnClick,
                           String negativeLabel, DialogInterface.OnClickListener negativeOnClick, boolean isCancelAble){

        AlertDialog.Builder builder = new AlertDialog.Builder(activity);
        builder.setTitle(title);
        builder.setCancelable(isCancelAble);
        builder.setMessage(msg);
        builder.setPositiveButton(positiveLabel, positiveOnClick);
        builder.setNegativeButton(negativeLabel, negativeOnClick);

        AlertDialog alert = builder.create();
        alert.show();
    }

    public void showDialogScore(Activity activity, int img, String title, String msg, String positiveLabel,
                           DialogInterface.OnClickListener positiveOnClick,
                           String negativeLabel, DialogInterface.OnClickListener negativeOnClick, boolean isCancelAble){

        AlertDialog.Builder builder = new AlertDialog.Builder(activity);
        builder.setIcon(img);
        builder.setTitle(title);
        builder.setMessage(msg);
        builder.setCancelable(isCancelAble);
        builder.setPositiveButton(positiveLabel, positiveOnClick);
        builder.setNegativeButton(negativeLabel, negativeOnClick);

        AlertDialog alert = builder.create();
        alert.show();
    }
}

File: app/src/main/java/com/rendersoncs/reportform/view/services/util/AlertDialogUtil.kt
package com.rendersoncs.reportform.view.services.util

import android.app.Activity
import android.content.DialogInterface
import androidx.appcompat.app.AlertDialog

class AlertDialogUtil {
    fun showDialog(activity: Activity?, title: String?, msg: String?, positiveLabel: String?,
                   positiveOnClick: DialogInterface.OnClickListener?,
                   negativeLabel: String?, negativeOnClick: DialogInterface.OnClickListener?,
                   isCancelAble: Boolean) {
        val builder = AlertDialog.Builder(activity!!)
        builder.setTitle(title)
        builder.setCancelable(isCancelAble)
        builder.setMessage(msg)
        builder.setPositiveButton(positiveLabel, positiveOnClick)
        builder.setNegativeButton(negativeLabel, negativeOnClick)
        val alert = builder.create()
        alert.show()
    }

    fun showDialogScore(activity: Activity?, img: Int, title: String?, msg: String?,
                        positiveLabel: String?,
                        positiveOnClick: DialogInterface.OnClickListener?,
                        negativeLabel: String?, negativeOnClick: DialogInterface.OnClickListener?,
                        isCancelAble: Boolean) {
        val builder = AlertDialog.Builder(activity!!)
        builder.setIcon(img)
        builder.setTitle(title)
        builder.setMessage(msg)
        builder.setCancelable(isCancelAble)
        builder.setPositiveButton(positiveLabel, positiveOnClick)
        builder.setNegativeButton(negativeLabel, negativeOnClick)
        val alert = builder.create()
        alert.show()
    }
}